scripts/o3de/o3de/utils.py

```diff
@@ -319,3 +323,29 @@ def find_ancestor_dir_containing_file(target_file_name: pathlib.PurePath, start_
     """
     ancestor_file = find_ancestor_file(target_file_name, start_path, max_scan_up_range)
     return ancestor_file.parent if ancestor_file else None
+
+
+def remove_gem_duplicates(gems: list) -> list:
+    """
+    For working with the 'gem_names' lists in project.json
+    Adds names to a dict, and when a collision occurs, eject the existing one in favor of the new one.
+    This is because when adding gems the list is extended, so the override will come last.
+    :param gems: The original list of gems, strings or small dicts (json objects)
+    :return: A new list with duplicate gem entries removed
+    """
+    new_list = []
+    names = {}
+    for gem in gems:
+        if isinstance(gem, dict):
+            if gem['name'] not in names:
```

Comment:
It is safer to use `gem.get('name', '')` just in case the the user has a project.json with a "gem_names" object that doesn't have a "names" field in it.
i.e
```json
"gem_names": [
    "MyGem",
    {
        "name": "YourGem":
    },
    {
        "name_field_mismatch": "BadFieldGem":
    },
]
```
In that case the python code would raise an exception.

Reply:
Alright, I can do that.  It would address the added test you recommended below.